<launch>!

  <arg name="upper_camera_serial" default="171864"/>
  <arg name="lower_camera_serial" default="171865"/>

  <arg name="rate" default="10" doc="Frequenz of camera data request in seconds"/>

  <node ns="n35_upper_camera" pkg="ensenso_camera" type="ensenso_camera_node" name="ensenso_camera" output="screen">
    <param name="serial" value="$(arg upper_camera_serial)"/>
    <param name="camera_frame" value="ensenso_upper_optical_frame"/>
    <param name="target_frame" value="base_link"/>  <!-- Frame in which we want x,y,z coodinates -->
    <param name="settings" value="$(find invite_vision)/config/ensenso_conf_171864_fast.json"/>
  </node>

  <node ns="n35_lower_camera" pkg="ensenso_camera" type="ensenso_camera_node" name="ensenso_camera" output="screen">
    <param name="serial" value="$(arg lower_camera_serial)"/>
    <param name="camera_frame" value="ensenso_lower_optical_frame"/>
    <param name="target_frame" value="base_link"/>  <!-- Frame in which we want x,y,z coodinates -->
    <param name="settings" value="$(find invite_vision)/config/ensenso_conf_171865_fast.json"/>    
  </node>

  <node pkg="invite_vision" type="request_upper_camera_data" name="request_upper_camera_data" output="screen">
    <param name="rate" value="$(arg rate)"/>
  </node>

  <node pkg="invite_vision" type="request_lower_camera_data" name="request_lower_camera_data" output="screen">
    <param name="rate" value="$(arg rate)"/>
  </node>


</launch>
